oauthproxy.go

```diff
@@ -297,6 +301,11 @@ func NewOAuthProxy(opts *options.Options, validator func(string) bool) *OAuthPro
 
 	logger.Printf("Cookie settings: name:%s secure(https):%v httponly:%v expiry:%s domains:%s path:%s samesite:%s refresh:%s", opts.Cookie.Name, opts.Cookie.Secure, opts.Cookie.HTTPOnly, opts.Cookie.Expire, strings.Join(opts.Cookie.Domains, ","), opts.Cookie.Path, opts.Cookie.SameSite, refresh)
 
+	extraHeaders := make([][]string, len(opts.ExtraHeaders))
+	for i, extraHeader := range opts.ExtraHeaders {
+		match := headerSplitRegex.FindStringSubmatch(extraHeader)
+		extraHeaders[i] = []string{match[1], match[2]}
+	}
```

Comment:
What happens if it doesn't match?

Reply:
According to the docs, it looks like match is nil in that case. So it will crash with a bad header specified. Does a sanity check go here or elsewhere?

Reply:
Now that it's in the validation, we shouldn't have an issue here, but I would suggest we still have a check here

```
if match == nil {
  return nil, fmt.Errorf("error parsing extra header: %s", extraHeader)
}
```

Reply:
can't do that. it only has one argument. I can have it return nil though.